Controllers/ImageRecognizer.cs

```diff
@@ -1,36 +1,123 @@
-
-            foreach (var imgPath in dir.GetFiles("r1.jpg"))
+            foreach (var imgPath in dir.GetFiles("*.jpg"))
             {
-                Debug.WriteLine(imgPath);
-
-                using var ocrEngineBest = new TesseractEngine(TessDataPath, "lit", EngineMode.Default);
-                //ocrEngineBest.SetVariable("tessedit_char_whitelist", "0123456789,.");
-
-                using var img = Pix.LoadFromFile(imgPath.ToString());
-                var imgGray = img.ConvertRGBToGray();
-
-                var res = ocrEngineBest.Process(imgGray).GetText();
+                var res = Recognize(imgPath);
 
                 var destPath = imgPath.Directory.Parent.FullName + @"\receiptsTxt\" + imgPath.Name + ".txt";
                 Debug.WriteLine(destPath);
-                
+
                 File.WriteAllText(destPath, res);
             }
         }
+
+        private string Recognize(FileInfo imagePath)
+        {
+            Debug.WriteLine(imagePath);
+
+            // Original Image
+            using var originalImage = Image.FromFile(imagePath.FullName);
+            double scaleIndex = (double) originalImage.Width / originalImage.Height;
+
+            // Auto crop Image
+
+            // Scaled Image
+            var resizedImage = ResizeImage(originalImage, 500, (int)(500 / scaleIndex));
+
+            return ProccessImage(resizedImage);
+
+            // Split Image
+            // SplitImage(resizedImage);
+        }
+
+        /// <summary>
+        ///     Resize the image to the specified width and height.
+        /// </summary>
+        /// <param name="image">The image to resize.</param>
+        /// <param name="width">The width to resize to.</param>
+        /// <param name="height">The height to resize to.</param>
+        /// <returns>The resized image.</returns>
+        private Bitmap ResizeImage(Image image, int width, int height)
+        {
+            var destRect = new Rectangle(0, 0, width, height);
+            var destImage = new Bitmap(width, height);
+
+            destImage.SetResolution(image.HorizontalResolution, image.VerticalResolution);
+
+            using (var graphics = Graphics.FromImage(destImage))
+            {
+                graphics.CompositingMode = CompositingMode.SourceCopy;
+                graphics.CompositingQuality = CompositingQuality.HighQuality;
+                graphics.InterpolationMode = InterpolationMode.HighQualityBicubic;
+                graphics.SmoothingMode = SmoothingMode.HighQuality;
+                graphics.PixelOffsetMode = PixelOffsetMode.HighQuality;
+
+                using var wrapMode = new ImageAttributes();
+                wrapMode.SetWrapMode(WrapMode.TileFlipXY);
+                graphics.DrawImage(image, destRect, 0, 0, image.Width, image.Height, GraphicsUnit.Pixel, wrapMode);
+            }
+
+            return destImage;
+        }
+
+        private IEnumerable<Image> SplitImage(Image image)
```

Comment:
Who is this method for? Just to return List <image>?

Reply:
Maybe I will implement it later on, because it can enhance text recognition, if image will be split in smaller parts.
But now it just does nothing.